packages/react/dropdown-menu/src/DropdownMenu.tsx

```diff
@@ -23,18 +25,32 @@ type DropdownMenuContextValue = {
   onOpenToggle(): void;
 };
 
+const SubmenuContext = React.createContext<boolean | undefined>(undefined);
 const [DropdownMenuProvider, useDropdownMenuContext] = createContext<DropdownMenuContextValue>(
   DROPDOWN_MENU_NAME
 );
 
-type DropdownMenuOwnProps = {
+type DropdownMenuOwnProps = React.ComponentProps<typeof DropdownMenuImpl>;
+
+const DropdownMenu: React.FC<DropdownMenuOwnProps> = (props) => {
+  const parentMenuContext = React.useContext(SubmenuContext);
+  return (
+    <SubmenuContext.Provider value={parentMenuContext === undefined ? false : true}>
```

Comment:
Why don't we do the same here then and just move this context to `DropdownMenuContent`... that way we're only a submenu if rendered inside content?

```jsx
<SubmenuContext.Provider value={true}>
```

I can't see an `const isRoot = parentContext === undefined` check anywhere in dropdown menu hence my confusion as to why we need `parentMenuContext` to be `undefined`.

Reply:
In the previous config we needed to have an additional state available for the first render to know if the context even exists, hence the `undefined`, but wrapping the content means we can do it after that point, so we implicitly know it's root. https://github.com/radix-ui/primitives/pull/629/commits/f9061af8d8f45b4224c005daa0212a22a66b061f